Lombiq.Hosting.MediaTheme.Deployer/Program.cs

```diff
@@ -0,0 +1,172 @@
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.Globalization;
+using System.IO.Compression;
+using System.Reflection;
+using static System.Console;
+
+namespace Lombiq.Hosting.MediaTheme.Deployer;
+
+internal static class Program
+{
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
+        "Globalization",
+        "CA1303:Do not pass literals as localized parameters",
+        Justification = "It's a console application it doesn't need localization.")]
+    public static void Main(string[] args)
+    {
+        if (args.Length == 3 && bool.TryParse(args[2], out bool clearMediaFolder))
+        {
+            // Creating directory for the deployment.
+            var currentRootDirectory = Directory.GetDirectoryRoot(Directory.GetCurrentDirectory());
+            var newDirectoryPath = Path.Join(currentRootDirectory, "MediaThemeDeployment_")
+                + DateTime.Now.ToString("ddMMMyyyyHHmmss", CultureInfo.CurrentCulture);
+
+            try
+            {
+                // Determine whether the directory exists.
+                if (Directory.Exists(newDirectoryPath))
+                {
+                    WriteLine("That file already exists.");
+                    return;
+                }
+
+                // Try to create the directory.
+                Directory.CreateDirectory(newDirectoryPath);
+
+                WriteLine("The directory was created successfully. {0}", newDirectoryPath);
+            }
+            catch (Exception exception)
+            {
+                WriteLine("The directory creation failed: {0}", exception.ToString());
+                return;
+            }
+
+            var pathToTheme = args[0];
+
+            // Creating media theme step.
+            dynamic mediaThemeStep = new JObject();
+            mediaThemeStep.name = "mediatheme";
+            mediaThemeStep.BaseThemeId = args[1];
+            mediaThemeStep.ClearMediaThemeFolder = clearMediaFolder;
+
+            // Creating media step.
+            var files = new JArray();
+
+            // Getting assets.
+            var pathToAssets = Path.Join(pathToTheme, "wwwroot");
+
+            var allAssetsPaths = Directory.EnumerateFiles(pathToAssets, "*", SearchOption.AllDirectories);
+
+            foreach (var assetPath in allAssetsPaths)
+            {
+                dynamic assetJObject = new JObject();
+                assetJObject.SourcePath = Path.Join(
+                    "_MediaTheme/Assets",
+                    assetPath[pathToAssets.Length..].Replace("\\", "/"));
+                assetJObject.TargetPath = assetJObject.SourcePath;
+
+                files.Add(assetJObject);
+            }
+
+            // Copying assets to deployment directory.
+            CopyDirectory(pathToAssets, Path.Join(newDirectoryPath, "\\_MediaTheme\\Assets"));
+
+            // Getting templates.
+            var pathToTemplates = Path.Join(pathToTheme, "Views");
+
+            var allTemplatesPaths = Directory
```

Comment:
When copying the files and adding them to the recipe, please do a string replace in the filename: change `.` to `_` and `-` to `__`.